Add Async MySQL support

vdauchy opened the pull request:
Hi @mpociot ,

This PR is a follow up to https://github.com/beyondcode/laravel-websockets/pull/644 discussion.

In the PR I did:
    - Add `react/mysql` to composer to support async connection to MySql.
    - Reformatter the config for database connections: This is a proposition to support a broader, less specific, way to configure DB connections that shall match Laravel's way of configure databases.
    - Add singleton on Async MySql connection.
    - Add deferred rejection callback in both SQLite and MySql AppManagers, the reason would be able to log errors properly (like connection errors, no table, etc...). Unfortunatelly I did not see any usable upstream logger, maybe it's something missing ?
 
Testing on a personnal project this all seem to work as expected.

mpociot commented:
Awesome work on this - thank you so much for the PR!
Can you modify the database configuration array so that it has an array for each driver? Pretty much like what Laravel does. I think this would be the best option and also not confuse users. 

Regarding the logger: yeah I don’t think there’s anything available yet, besides logging to CLI.

vdauchy commented:
Hi @mpociot,

I corrected the db configuration structure, I removed the the checks on singletons so now both sqlite and mysql singleton are registered everytime.

An other way database configuration could be handled could be by setting the connection name (configured in Laravel's database.php file) instead. That would limit duplication of configurations, and one of the big point I see: Async db connection could be only used on the WS process, for UI since you go through the Laravel app (Nginx, Apache, etc...) you may just use a classic Eloquent model... So both connection may surely be duplicated at the end.

If we even want to push the idea further, we could even just gives a model class in the config and then get the db connection from there, then replace stuff like ```database->query('SELECT * from `apps` WHERE `secret` = ?', [$appSecret])``` by something like: ```database->query($appModel::query()->whereSecret(null)->toSql(), [$appSecret])```...

But that's just brain food... :-)

Regarding the MySQL tests, I will need to dig on MySQL github action I suppose... will be fun :-)

mpociot commented:
@vdauchy great - thank you!

Hm..I guess it all depends on how flexible we want this to be.
We could also just make a database connection configurable and then pull the settings from the `database.php` config file.
This would completely avoid duplicating configurations. 
Of course, this would only work with supported drivers, but we can list the ones we support out of the box in the comment + documentation.
So maybe this would be even easier and user-friendly.

Providing a model would work as well, but I feel that this might be confusing. Which model should you pick? Because it's not really about the model, but the database connection.
